### PR TITLE
Add instance choice classes to traitsui.api

### LINKED ISSUE #1494: InstanceChoice, InstanceChoiceItem, and InstanceFactory should be added to traitsui.api
These classes are part of the _secret_ `InstanceEditor` api, but they should not be so secret!  They are expected to need to be used along with the `InstanceEditor` for certain use cases and thus should be exposed in the api module.

### PR DESCRIPTION
**Pull request by aaronayres35**
closes #1494 

This PR simply adds instance choice related classes to the `traitsui.api`

### CODE REVIEW

File: traitsui/api.py [hunk 1]
```diff
@@ -159,6 +159,14 @@
...
 - :class:`~.TableFilter`
 
+Instance UI
+-----------
+
+- :class:`~.InstanceChoice`
+- :class:`~.InstanceChoiceItem`
+- :class:`~.InstanceDropChoice`
+- :class:`~.InstanceFactoryChoice`
+
```
Comment (rahulporuri) [inline traitsui/api.py:169]: Let's move the `InstanceEditor` down here, it'll be cleaner.

**Review by rahulporuri — APPROVED** [review]
LGTM

File: traitsui/api.py (hunk 1 above)
Comment (aaronayres35) [reply traitsui/api.py:169]: There are comparable sections for "Tree UI" and "Table UI", but the `TreeEditor` and `TableEditor` are still included in the full list with the other editors not in those sections.  I am still happy to make this suggested change, but wanted to point this out from a consistency stand point

Comment (rahulporuri) [reply traitsui/api.py:169]: In that case, let's move the `TreeEditor` and `TableEditor` to the Tree UI and Table UI sections respectively - along with the change I recommended earlier.

Alternatively, if we want to list all of the available editors in one location i.e. under the Editors section (how I think things are currently), we can choose to add duplicate information in the Table UI, Tree UI and Instance UI sections.

**Review by rahulporuri — APPROVED** [review]
Still LGTM. Can you check if there are any new sphinx warnings generated because of these changes when the documentation is built?

**Comment by aaronayres35** [comment]
> Still LGTM. Can you check if there are any new sphinx warnings generated because of these changes when the documentation is built?

Just checked and I'm not seeing any new warnings from these changes

**Pull request merged**